File: Travelo.Model/Agency.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace Travelo.Model
{
    public partial class Agency
    {
        public int Id { get; set; }
        public string Name { get; set; }
        public string Email { get; set; }
        public string Phone { get; set; }
        public string WebsiteUrl { get; set; } 
        public string About { get; set; }

        public byte[] Image { get; set; } = null;
        public bool Status { get; set; }
        public string Address { get; set; }
        public int CityId { get; set; }
        public string CityName { get; set; }
        public string PostalCode { get; set; }
        public string Location { get; set; }
        public int numberOfTrips { get; set; }
        public int rating { get; set; } = 5;
    }
}

File: travelo_agency/lib/models/agency.dart
import 'package:json_annotation/json_annotation.dart';

part 'agency.g.dart';

@JsonSerializable()
class Agency {
  int? id;
  String? name;
  String? email;
  String? phone;
  String? websiteUrl;
  String? about;
  String? image;
  bool? status;
  String? location;
  int? numberOfTrips;
  double? rating;
  String? address;
  int? cityId;
  String? cityName;
  String? postalCode;

  Agency();

  factory Agency.fromJson(Map<String, dynamic> json) => _$AgencyFromJson(json);

  Map toJson() => {
        "id": id,
        "name": name,
        "email": email,
        "phone": phone,
        "websiteUrl": websiteUrl,
        "about": about,
        "image": image,
        "status": status,
        "location": location,
        " numberOfTrips": numberOfTrips,
        " rating": rating,
        "address": address,
        " cityId": cityId,
        "cityName": cityName,
        "postalCode": postalCode,
      };
}
